packages/gator-permissions-snap/src/clients/blockchainMetadataClient.ts

```diff
@@ -327,4 +331,68 @@ export class BlockchainTokenMetadataClient implements TokenMetadataClient {
       );
     }
   }
+
+  /**
+   * Checks if a transaction receipt is valid by calling the eth_getTransactionReceipt method.
+   * @param args - The parameters for checking the transaction receipt.
+   * @param args.txHash - The hash of the transaction to check.
+   * @param args.chainId - The chain ID in hex format.
+   * @returns True if the transaction receipt is valid, false if it is not.
+   * @throws InvalidInputError if input parameters are invalid.
+   * @throws ChainDisconnectedError if the provider is on the wrong chain.
+   * @throws ResourceUnavailableError if the on-chain check fails and we cannot determine the status.
+   */
+  public async checkTransactionReceipt({
+    txHash,
+    chainId,
+  }: {
+    txHash: Hex;
+    chainId: Hex;
+  }): Promise<boolean> {
+    logger.debug('BlockchainTokenMetadataClient:checkTransactionReceipt()', {
+      txHash,
+      chainId,
+    });
+
+    if (!chainId) {
+      const message = 'No chain ID provided';
+      logger.error(message);
+      throw new InvalidInputError(message);
+    }
+
+    await ensureChain(this.#ethereumProvider, chainId);
+
+    try {
+      const result = await getTransactionReceipt({
+        ethereumProvider: this.#ethereumProvider,
+        txHash,
+        isRetryableError: (error) => this.#isRetryableError(error),
```

Comment:
nit:
```suggestion
        isRetryableError: this.#isRetryableError,
```

The impact is small, but this reduces the callstack depth